egs/librispeech/ASR/pruned_transducer_stateless2/conformer.py

```diff
@@ -117,17 +151,148 @@ def forward(
             # Caution: We assume the subsampling factor is 4!
             lengths = ((x_lens - 1) // 2 - 1) // 2
         assert x.size(0) == lengths.max().item()
-        mask = make_pad_mask(lengths)
 
-        x = self.encoder(
-            x, pos_emb, src_key_padding_mask=mask, warmup=warmup
+        src_key_padding_mask = make_pad_mask(lengths)
+        mask = None
+
+        if self.dynamic_chunk_training:
+            assert (
+                self.causal
+            ), "Causal convolution is required for streaming conformer."
+            max_len = x.size(0)
+            chunk_size = torch.randint(1, max_len, (1,)).item()
```

Comment:
The method to choose `chunk_size` here is from wenet's paper, we may tune it to find a better one in the future.